rtbkit/common/exchange_connector.h

```diff
@@ -231,6 +231,30 @@ struct ExchangeConnector: public ServiceBase {
     getCreativeCompatibility(const Creative & creative,
                              bool includeReasons) const;
 
+    /** Structure is used to determine if the bid is valid.
+    */
+    struct BidValidity {
+        BidValidity()
+            :isValidbid(true)
+        {
+        }
+
+        void setInvalidBid() {
```

Comment:
The reason should be configurable. `setInvalidBid` should take a `std::string` or `const char *` as a parameter so that the user can specify the reason.

`BidValidity` could maybe just inherit from `ExchangeCompatibility`.